ec/src/models/short_weierstrass/group.rs

```diff
@@ -216,34 +237,45 @@ impl<P: SWCurveConfig> Group for Projective<P> {
             let yy = self.y.square();
 
             // YYYY = YY^2
-            let mut yyyy = yy.square();
+            let mut yyyy = yy;
+            yyyy.square_in_place();
 
             // ZZ = Z1^2
-            let zz = self.z.square();
+            let mut zz = self.z;
+            zz.square_in_place();
 
             // S = 2*((X1+YY)^2-XX-YYYY)
             let s = ((self.x + &yy).square() - &xx - &yyyy).double();
 
             // M = 3*XX+a*ZZ^2
-            let m = xx + xx.double() + P::mul_by_a(zz.square());
+            let mut m = xx;
+            m.double_in_place();
+            m += &xx;
+            m += &P::mul_by_a(zz.square());
 
             // T = M^2-2*S
-            let t = m.square() - &s.double();
-
             // X3 = T
-            self.x = t;
-            // Y3 = M*(S-T)-8*YYYY
-            let old_y = self.y;
-            self.y = m * &(s - &t) - &*yyyy.double_in_place().double_in_place().double_in_place();
+            self.x = m;
+            self.x.square_in_place();
+            self.x -= s.double();
+
             // Z3 = (Y1+Z1)^2-YY-ZZ
-            self.z = (old_y + &self.z).square() - &yy - &zz;
+            self.z *= self.y;
+            self.z.double_in_place();
```

Comment:
No longer the same formula?  The one here is (Y1 * Z1) * 2

Reply:
Is it "Can alternatively be computed as (Z1+H)^2-Z1Z1-HH, but the latter is slower.", as the case below?